Arm/State.lean

```diff
@@ -680,52 +686,61 @@ As an easy corollary, we get `getLsb` theorems on these to allow
 `omega` based reasoning about bit-level values of memory.
 -/
 
-abbrev Memory := Store (BitVec 64) (BitVec 8)
+def Memory.read (addr : BitVec 64) (m : Memory) : BitVec 8 :=
+  read_store addr m
+
+theorem ArmState.read_mem_eq_mem_read : read_mem addr s = s.mem.read addr := rfl
 
 /--
-A variant of `read_mem` that directly talks about writes to memory, instead of over the entire `ArmState`
+A variant of `write_mem` that directly talks about writes to memory, instead of over the entire `ArmState`
 -/
-def read_mem' (addr : BitVec 64) (m : Memory) : BitVec 8 :=
-  read_store addr m
+def Memory.write (addr : BitVec 64) (val : BitVec 8) (m : Memory) : Memory :=
+  write_store addr val m
 
-theorem read_mem_eq_read_mem' : read_mem addr s = read_mem' addr s.mem := rfl
+theorem ArmState.write_mem_eq_mem_write :  (write_mem addr val s).mem = s.mem.write addr val := rfl
 
-@[simp]
-theorem getLsb_read_mem' : (read_mem' addr s).getLsb i = (s addr).getLsb i :=
-  rfl
+namespace Memory
+
+/--
+A variant of `read_mem` that directly talks about writes to memory, instead of over the entire `ArmState`
+-/
```

Comment:
Should be moved above the definition of `Memory.read` above

Reply:
Done! thanks